src/hotspot/share/opto/addnode.cpp

```diff
@@ -786,6 +839,31 @@ Node* OrINode::Ideal(PhaseGVN* phase, bool can_reshape) {
        return new RotateRightNode(in(1)->in(1), shift, TypeInt::INT);
      }
   }
+  // major_progress() check postpones the transformation after loop optimization
+  if (can_reshape && !phase->C->major_progress() && Matcher::match_rule_supported(Op_BitfieldInsertI)) {
+    Node *andi = NULL;
+    Node *dst = in(1);
+    Node *src = in(2);
+    int offset = 0;
+    // Convert Or(v1, Shift(And(v2, 0xFF), shift)) into BitfieldInsert(dst, src, width, offset)
```

Comment:
This comment is incoherent, misleading and totally inadequate. What is the 'Or' there for? Why doe it describe an And nested in a Shift when the code expects other cases. This is no use whatsoever to anyone who has to maintain your code.